pkg/vpcmodel/grouping.go

```diff
@@ -306,6 +306,7 @@ func getKeyOfGroupConnLines(ep EndpointElem, connection string) string {
 // assuming the  g.groupedLines was already initialized by previous step groupExternalAddresses()
 func (g *GroupConnLines) groupInternalSrcOrDst(srcGrouping, groupVsi bool) {
 	res, groupingSrcOrDst := g.groupLinesByKey(srcGrouping, groupVsi)
+	res = g.unifiedGroupedConnLines(res)
```

Comment:
How is `res` affected by `extendGroupingSelfLoops`? It seems that only `groupLinesByKey` adds lines to `res`,  so why on this stage the func `unifiedGroupedConnLines` works on `res`?

Reply:
@adisos good catch. So there are 2 questions to answer here: yours, and why still the fix that seems misplaced solved the issue.
The original problem here is that `extendGroupingSelfLoops` "mess up with"  the basic result of the grouping, modifying both sides and not only the targeted one. This is why we had a problem in the first place. 
The fix I've put in indeed does not cover the problem, as you say. The reason it did seem to cover the problem is that is fixed the result `res` of the previous stage and for the case inline it was suffice. Clearly this is not a proper solution. I'll fix.

Reply:
fixed @haim-kermany please verify it is working as expected (by my testing it is)